### PR TITLE
deployment

### PR DESCRIPTION
**Pull request by bvavala**
Signed-off-by: Bruno Vavala <bruno.vavala@intel.com>

### CODE REVIEW

File: text/0000-fabric-private-chaincode-lite.md [hunk 1]
```diff
@@ -310,64 +310,72 @@ The registry is particularly relevant for clients, for retrieving an FPC chainco
...
+query the `initEnclave` FPC chaincode function;
+pass the result as an argument in the invocation to the `registerEnclave` Enclave Registry chaincode function.
+-->
+A successful initialization and registration will result in a new entry (the enclave credentials, which include the chaincode definition, attestation and key material) in the registry namespace on the ledger.
+
+Internally, the command operates as follows.
+First, it issues an `initEnclave` query which reaches the FPC Shim.
+The shim initializes the enclave (which will process FPC chaincode invocations) with the chaincode parameters received from the peer, namely: the chaincode definition and the channel identifier.
+Then, it generate the key material: public/private key-pairs for signing and encryption, and the state encryption key.
+The public signing key is used as enclave identifier.
```
Comment (g2flyer) [inline text/0000-fabric-private-chaincode-lite.md:365]: This is missing also the chaincode encryption key-pair. And note that in the (referenced) specs, this key-generation is via a separate `generateCCkeys` query, not in `initEnclave` and also a later corresponding ercc query...

File: text/0000-fabric-private-chaincode-lite.md [hunk 2]
```diff
@@ -310,64 +310,72 @@ The registry is particularly relevant for clients, for retrieving an FPC chainco
...
 
-	Once the enclave is up and running, the `peer lifecycle chaincode createenclave` command invokes a register transaction for the newly created Chaincode Enclave.
-	The public parameters of the Chaincode Enclave along with the attestation evidence is sent to the FPC Registry, which validates them internally by checking consistency and signatures, and then stores the new entry.
-	The register invocation is invoked at a sufficient set of endorsing peers to satisfy the FPC Registry endorsement policy.
-	Finally, a Register transaction is submitted to the Ordering Service.
-	By committing a Register transactions, the FPC Chaincode Attestation is now visible proof of validity of the FPC Chaincode to all participants in the Channel.
+The administrator of the peer hosting the FPC chaincode enclave is responsible for the initialization and registration of the enclave.
+The operation is performed by executing the `initEnclave` admin command.
+
+The command can be triggered through the FPC Client SDK.
```
Comment (g2flyer) [inline text/0000-fabric-private-chaincode-lite.md]:
```suggestion
The command can be triggered through the FPC Client SDK (see [earlier Deployment Section](#deployment)
```

**Review by g2flyer — CHANGES_REQUESTED** [review]

File: text/0000-fabric-private-chaincode-lite.md (hunk 2 above)
Comment (g2flyer) [reply text/0000-fabric-private-chaincode-lite.md]: Oops, too quick in pressing button, here now also the overall comment.  Looks overall good, just that the description of `initEnclave` admin command doesn't match the spec ...

File: text/0000-fabric-private-chaincode-lite.md (hunk 1 above)
Comment (bvavala) [reply text/0000-fabric-private-chaincode-lite.md:365]: It is a simplification, given that FPC Lite is designated-peer only, and that it is not distributing the key.

Comment (g2flyer) [reply text/0000-fabric-private-chaincode-lite.md:365]: That's not what our FPC Lite UML spec says (as refenced further down). That said, just occurred to me that the easiest way here is just not to talk about what exact flows happen between client, enclave and ercc, i.e., uplevel it a bit. But if we name flows, then they should be consistent with the UML

File: text/0000-fabric-private-chaincode-lite.md [hunk 3]
```diff
@@ -310,64 +310,72 @@ The registry is particularly relevant for clients, for retrieving an FPC chainco
...
 
-	Once the enclave is up and running, the `peer lifecycle chaincode createenclave` command invokes a register transaction for the newly created Chaincode Enclave.
-	The public parameters of the Chaincode Enclave along with the attestation evidence is sent to the FPC Registry, which validates them internally by checking consistency and signatures, and then stores the new entry.
-	The register invocation is invoked at a sufficient set of endorsing peers to satisfy the FPC Registry endorsement policy.
-	Finally, a Register transaction is submitted to the Ordering Service.
-	By committing a Register transactions, the FPC Chaincode Attestation is now visible proof of validity of the FPC Chaincode to all participants in the Channel.
+The administrator of the peer hosting the FPC chaincode enclave is responsible for the initialization and registration of the enclave.
+The operation is performed by executing the `initEnclave` admin command.
+
+The command can be triggered through the FPC Client SDK (see [earlier Deployment Section](#deployment)
```
Comment (g2flyer) [inline text/0000-fabric-private-chaincode-lite.md]:
```suggestion
The command can be triggered through the FPC Client SDK (see [earlier Deployment Section](#deployment)).
```

File: text/0000-fabric-private-chaincode-lite.md (hunk 1 above)
Comment (bvavala) [reply text/0000-fabric-private-chaincode-lite.md:365]: I vote for uplevel, suggestions?

Comment (mbrandenburger) [reply text/0000-fabric-private-chaincode-lite.md:365]: Let me merge this, do a pass, and create a new PR

**Pull request merged**